source/dofs/dof_handler_policy.cc

```diff
@@ -51,6 +51,64 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+namespace internal
+{
+  namespace p4est
+  {
+    template <>
+    struct types<1>
+    {
+      using quadrant = int;
+    };
+
+    template <>
+    bool
+    quadrant_is_equal<1>(const typename types<1>::quadrant &q1,
+                         const typename types<1>::quadrant &q2)
+    {
+      return q1 == q2;
+    }
+
+    template <>
+    bool quadrant_is_ancestor<1>(types<1>::quadrant const &q1,
+                                 types<1>::quadrant const &q2)
+    {
+      int gen_1 = (q1 << 27) >> 27;
+      int gen_2 = (q2 << 27) >> 27;
+
+      if (gen_1 >= gen_2)
+        return false;
+
+      int val_1 = (q1 >> (31 - gen_1)) << (31 - gen_1);
+      int val_2 = (q2 >> (31 - gen_1)) << (31 - gen_1);
```

Comment:
I suspect that this implementation actually does something useful, but because of a lack of commentary, I don't know what this is :-) Can you give 27 and 31 symbolic names and add a couple of lines of commentary? Same for the function below.